rust/sdk-processor/src/steps/common/gcs_uploader.rs

```diff
@@ -100,6 +61,10 @@ pub fn create_new_writer(schema: Arc<Type>) -> anyhow::Result<SerializedFileWrit
     SerializedFileWriter::new(Vec::new(), schema, props_arc).context("Failed to create new writer")
 }
 
+fn determine_parquet_type(buffer: &ParquetTypeStructs) -> ParquetTypeEnum {
```

Comment:
if you downcast and need the type still, I assume dispatch is better?

Reply:
agreed, I think enum dispatch is the better choice here